examples/preprocessing/plot_pace_classification.py

```diff
@@ -0,0 +1,275 @@
+"""
+PACE algorithm for classification analysis
+=======================================================================
+
+Explores the possibility to apply clustering techniques to sparse,
+irregularly sampled data using PACE.
+"""
+
+# Author: Alejandro Arias Gomez
+# License: MIT
+
+# sphinx_gallery_thumbnail_number = 5
+
+# %%
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from sklearn.utils import Bunch
+
+from skfda.datasets._real_datasets import fetch_country_height
+from skfda.preprocessing.dim_reduction import PACE
+from skfda.representation import FDataIrregular
+
+# %%
+# This example explores the possibility to apply clustering techniques to
+# sparse, irregularly sampled data using the PACE algorithm, as described in
+# :footcite:ts`yao+muller+wang_2005_pace`.
+#
+# Throughout this analysis, we will use the Country Height dataset, which
+# contains the average height of 144 countries grouped by decades, from 1810 to
+# 1989. The dataset is irregularly sampled, meaning that not all countries have
+# measurements for all decades.
+#
+# The dataset is available in the `skfda.datasets` module.
+#
+# Our goal is to analyse the relationship between the countries and their
+# continent using classification techniques. However, because the dataset is
+# irregularly sampled, we will use the PACE algorithm to perform FPCA analysis
+# and reconstruct the underlying curves for each country, converting the data
+# into a regular grid before classification, in order to be able to use the
+# package's algorithms.
+#
+# We will first load the dataset and plot each country's height curve, divided
+# by continent. In addition, we will represent how many countries we have per
+# continent, in order to compare the classification results with a
+# classificator by majority.
+country_height_bunch: Bunch = fetch_country_height()
+country_height: FDataIrregular = country_height_bunch.data
+assert isinstance(
+    country_height, FDataIrregular
+), "Expected an FDataIrregular object"
+
+target = country_height_bunch.target
+
+country_categories = country_height_bunch.target_names.categories
+n_groups = len(country_categories)
+cmap = plt.get_cmap("Set2")
```

Comment:
Why using a color map (which is for continuous gradients) instead of a color cycle? And why is the default color cycle not good enough?